File: VB/09_Interaction/AddImageInHeader.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace AddImageInHeader
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim path As String = "..\..\..\..\..\..\Data\"
			'pdf file 
			Dim input As String = path & "Sample4.pdf"

			'open pdf document
			Dim doc As New PdfDocument(input)

			'create a pdf image
			Dim headerImage As PdfImage = PdfImage.FromFile(path & "E-iceblue logo.png")

			'draw header image into pages
			For Each page As PdfPageBase In doc.Pages
				page.Canvas.DrawImage(headerImage, New PointF(10, 2))
			Next page

			Dim output As String = "AddImageInHeader.pdf"
			'Save pdf file.
			doc.SaveToFile(output)
			doc.Close()

			'Launching the Pdf file.
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Interaction/AddImageInHeader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;

namespace AddImageInHeader
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string path = "..\\..\\..\\..\\..\\..\\Data\\";
            //pdf file 
            string input = path + "Sample4.pdf";

            //open pdf document
            PdfDocument doc = new PdfDocument(input);

            //create a pdf image
            PdfImage headerImage = PdfImage.FromFile(path + "E-iceblue logo.png");

            //draw header image into pages
            foreach (PdfPageBase page in doc.Pages)
            {
                page.Canvas.DrawImage(headerImage, new PointF(10, 2));
            }

            string output = "AddImageInHeader.pdf";
            //Save pdf file.
            doc.SaveToFile(output);
            doc.Close();

            //Launching the Pdf file.
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
